clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -5294,24 +5294,22 @@ static void handleNoGlobalWorkOffsetAttr(Sema &S, Decl *D,
 
   checkForDuplicateAttribute<SYCLIntelNoGlobalWorkOffsetAttr>(S, D, Attr);
 
-  uint32_t Enabled = 1;
-  if (Attr.getNumArgs()) {
-    const Expr *E = Attr.getArgAsExpr(0);
-    if (!checkUInt32Argument(S, Attr, E, Enabled, 0,
-                             /*StrictlyUnsigned=*/true))
-      return;
-  }
-  if (Enabled > 1)
-    S.Diag(Attr.getLoc(), diag::warn_boolean_attribute_argument_is_not_valid)
-        << Attr;
-
   if (Attr.getKind() == ParsedAttr::AT_SYCLIntelNoGlobalWorkOffset &&
       checkDeprecatedSYCLAttributeSpelling(S, Attr))
     S.Diag(Attr.getLoc(), diag::note_spelling_suggestion)
         << "'intel::no_global_work_offset'";
 
-  D->addAttr(::new (S.Context)
-                 SYCLIntelNoGlobalWorkOffsetAttr(S.Context, Attr, Enabled));
+  // If no attribute argument is specified, set to default value '1'.
+  if (!Attr.isArgExpr(0)) {
+    Expr *E = IntegerLiteral::Create(S.Context, llvm::APInt(32, 1),
+                                     S.Context.IntTy, Attr.getLoc());
+    D->addAttr(::new (S.Context)
+                   SYCLIntelNoGlobalWorkOffsetAttr(S.Context, Attr, E));
+  } else {
```

Comment:
Why two separate code paths? I think it'd be more clear to do:
```
Expr *E = Attr.isArgExpr(0) ? Attr.getArgAsExpr(0) : IntegerLiteral::Create(...);
D->addAttr(...);
```

Reply:
@AaronBallman for looking into the patch and feedback. 
Yes, no need to have a separate code paths. I have updated it.

Reply:
Done